src/v/cluster/rm_stm.cc

```diff
@@ -1464,26 +1465,28 @@ void rm_stm::abort_old_txes() {
     });
 }
 
+fragmented_vector<model::producer_identity>
+rm_stm::get_expired_producers() const {
+    fragmented_vector<model::producer_identity> result;
+    auto pids = std::views::keys(_log_state.ongoing_map);
```

Comment:
are we certain that using `std::ranges` is now blessed ?

Reply:
Judging from https://libcxx.llvm.org/Status/Cxx20.html ranges support should be good in clang 16.

Reply:
Though I don't find this any more readable than a for loop :)

Reply:
I see references to ranges elsewhere in the code.. I agree it is not any more readable than a range for but opens up possibilities to use std functions like find_if and such.